src/clue.js

```diff
@@ -2,25 +2,214 @@
+        occupation: 'Actor',
+        age: 31,
+        description: 'She is an A-list movie star with a dark past',
+        image: 'https://www.radiotimes.com/uploads/images/Original/111967.jpg',
+        color: 'red'
+      }, 
+      
+      {
+        firstName: 'Eleanor',
+        lastName: 'Peacock',
+        occupation: 'Socialité',
+        age: 36,
+        description: 'She is from a wealthy family and uses her status and money to earn popularity',
+        image: 'https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg',
+        color: 'blue'
+      }, 
+      
+      {
+        firstName: 'Jack',
+        lastName: 'Mustard',
+        occupation: 'Retired Football player',
+        age: 62,
+        description: 'He is a former football player who tries to get by on his former glory',
+        image: 'https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg',
+        color: 'yellow'
+      }];
 
 // Rooms Array
 
-const roomsArray = [];
+const roomsArray = [
+
+    {
+        name: 'Dining Room',
+    },
+
+    {
+        name: 'Conservatory',
+    },
+
+    {
+        name: 'Kitchen',
+    },
+
+    {
+        name: 'Study',
+    },
+
+    {
+        name: 'Library',
+    },
+
+    {
+        name: 'Billiard Room',
+    },
+
+    {
+        name: 'Lounge',
+    },
+
+    {
+        name: 'Ballroom',
+    },
+
+    {
+        name: 'Hall',
+    },
+
+    {
+        name: 'Spa',
+    },
+
+    {
+        name: 'Living Room',
+    },
+
+    {
+        name: 'Observatory',
+    },
+
+    {
+        name: 'Theater',
+    },
+
+    {
+        name: 'Guest House',
+    },
+
+    {
+        name: 'Patio',
+    }];
 
 // Weapons Array
 
-const weaponsArray = [];
+const weaponsArray = [
+
+    {
+       name: 'rope',
+       weight: 10,
+    },
+
+    {
+        name: 'knife',
+        weight: 8,
+    },
+
+    {
+        name: 'candlestick',
+        weight: 2,
+    },
+
+    {
+        name: 'dumbbell',
+        weight: 30,
+    },
+
+    {
+        name: 'poison',
+        weight: 2,
+    },
+
+    {
+        name: 'axe',
+        weight: 15,
+    },
+
+    {
+        name: 'bat',
+        weight: 13,
+    },
+
+    {
+        name: 'trophy',
+        weight: 25,
+    },
+
+    {
+        name: 'pistol',
+        weight: 20,
+    }];
 
 
 // ITERATION 2
 
-function selectRandom() {}
+function selectRandom(randomArray) {
+
+    // get random index value
+    const randomIndex = Math.floor(Math.random() * randomArray.length);
+
+    // get random element
+    const randomElement = randomArray[randomIndex];
 
-function pickMystery() {}
+    return randomElement;
+}
 
+function pickMystery() {
 
+    return {
+        
+        suspect: selectRandom(suspectsArray), 
+        weapon: selectRandom(weaponsArray),
+        room: selectRandom(roomsArray)
+    };
+
+}
 // ITERATION 3
 
-function revealMystery() {}
+function revealMystery() {
+
+    const mystery = pickMistery();
```

Comment:
the reason why the test is not passing is that revealMystery shouldn't create a new mystery but get an already existing mysteryObject as input. 

```javascript
function revealMystery(mystery) {
      return `${mystery.firstName} ${mystery.lastName} killed Mr. Boddy using the ${mystery.weapon} in the ${mystery.place}!`;
} 
``` 
should do the trick